Implement CallHost and CallIndirect opcodes

Leonardo2718 opened the pull request:
This changeset implements the following opcodes:

- `InterpCallHost`
- `CallIndirect`

Closes #53 
Closes #75

aviansie-ben commented on src/jit/function-builder.cc:
```diff
@@ -132,6 +156,32 @@ FunctionBuilder::Result_t FunctionBuilder::CallHelper(wabt::interp::Thread* th,
+  using namespace wabt::interp;
+  auto* env = th->env_;
+  Table* table = &env->tables_[table_index];
+  Index entry_index = th->Pop<uint32_t>();
```
This line is requiring a lot of ugly hacks to be brought in. Would it be possible to pop this value as an `i32` inside the JITted code and then pass it into the helper as a parameter to avoid this?

Leonardo2718 commented:
@aviansie-ben I followed your suggestion and changed the `CallIndirect` implementation to generated IL directly for the Pop and pass the value to the `CallIndirectHelpder`.

I've also added a copy of the interpreter test for `CallIndirect` opcode to the JIT tests. As I note in the commit comment, it only provides minimal amounts of testing as it does not cover all cases the JIT must handle. However, I hope it will be sufficient for the time being.

Once this PR is merged, we should open an issue to add more testing for both the `CallIndirect` and `InterpCallHost` functions.
